### PR TITLE
perf: Do not copy dataframe if uploading its index

### PR DESCRIPTION
**Pull request by jasonni0**
Not sure why df.copy() used. reset_index() should create the index column.

### CODE REVIEW

**Comment by jasonni0** [comment]
Test case (271360 rows x 8 columns):
- df.reset_index() 0.0419158935546875 seconds
- df.copy(deep=True).reset_index() 0.07775378227233887 seconds

**Comment by borchero** [comment]
I'm wondering though, don't you need `drop=True` in the `reset_index` method?

**Comment by jasonni0** [comment]
> I'm wondering though, don't you need `drop=True` in the `reset_index` method?

`drop=True` in the `reset_index` will drop index column. You need index column if index flag is true. Check the code below.

Screenshot 2023-05-26 122353

**Comment by borchero** [comment]
Ah sorry, I wasn't paying particular attention where this is applied, I always have `index=False` in mind since the index is rarely useful 😅

**Comment by jasonni0** [comment]
Anyway I think `df = df.copy(deep=True).reset_index()` and `df = df.reset_index()` should be the same. I don't see `df.copy(deep=True)` is useful here, just copy a df and perform the same `reset_index()` function.

File: bcpandas/main.py [hunk 1]
```diff
@@ -410,7 +410,7 @@ def to_sql(
 
     # TODO diff way to implement? could be big performance hit with big dataframe
```
Comment (borchero) [inline bcpandas/main.py:411]:
```suggestion
```

**Comment by yehoshuadimarsky** [comment]
Adding context here - the reason I originally used `.copy()` is because I didn't want to mutate the inputs. I wasn't sure if `.reset_index()` called on the input would mutate it, which is definitely a bad thing. Inputs should not be mutated unless this is communicated or documented to the user.

A quick test just now shows that it does not mutate it, but I hope that's enough to verify this.

```ipython
In [1]: import pandas as pd, numpy as np

In [2]: pd.__version__
Out[2]: '1.5.2'

In [3]: np.__version__
Out[3]: '1.23.5'

In [4]: df = pd.DataFrame(np.random.rand(5,5))

In [5]: def f(frame):
   ...:     x = frame.reset_index()
   ...:     return x
   ...:

In [6]: f(df)
Out[6]:
   index         0         1         2         3         4
0      0  0.182830  0.426792  0.748639  0.555703  0.928144
1      1  0.120262  0.552670  0.219774  0.075147  0.974727
2      2  0.363123  0.047610  0.951297  0.224003  0.518281
3      3  0.514103  0.933547  0.066933  0.820800  0.535219
4      4  0.006199  0.136653  0.827070  0.635302  0.550694

In [7]: df
Out[7]:
          0         1         2         3         4
0  0.182830  0.426792  0.748639  0.555703  0.928144
1  0.120262  0.552670  0.219774  0.075147  0.974727
2  0.363123  0.047610  0.951297  0.224003  0.518281
3  0.514103  0.933547  0.066933  0.820800  0.535219
4  0.006199  0.136653  0.827070  0.635302  0.550694
```

**Comment by borchero** [comment]
@yehoshuadimarsky I'm quite confident that `reset_index` doesn't modify the input. Afaik, that's the case with all dataframe methods which return a dataframe.

**Pull request merged**